### PR TITLE
added upgrade note for deprecated container property

### PR DESCRIPTION
**Pull request by ahmetakbn**
I am targetting this branch, because it is upgrade not of this branch.

refs #1035

### CODE REVIEW

**Comment by greg0ire** [comment]
> it is upgrade not of this brunch.

Makes me hungry ;)

**Comment by core23** [comment]
We don't need this, because the property is allready marked as `@deprecated`.

**Comment by greg0ire** [comment]
> We don't need this, because the property is allready marked as @deprecated.

Is this enough? We are providing deprecation messages, but for each of them, we also have an upgrade note, right?

**Comment by core23** [comment]
IMO, we have enough information for developers. It is marked as `@deprecated` and we throw a `trigger_error` for this. The Upgrade note for the next major release also have short [note](https://github.com/sonata-project/SonataMediaBundle/blob/master/UPGRADE-4.0.md#deprecations).

**Comment by ahmetakbn** [comment]
Was a requested by @Soullivaneuh FYI
https://github.com/sonata-project/SonataMediaBundle/pull/1035#issuecomment-234549024

And it is also written here under the Deprecated title : https://github.com/sonata-project/SonataMediaBundle/releases/tag/3.1.0

**Comment by greg0ire** [comment]
Let's reopen and assign this to @Soullivaneuh , ok?

**Comment by core23** [comment]
@Soullivaneuh should we add this note to the `UPGRADE-3.x.MD` file?

**Comment by Soullivaneuh** [comment]
> @Soullivaneuh should we add this note to the UPGRADE-3.x.MD file?

It was always like this. All deprecation must be documented.

BTW, this should not be just be a line with which is deprecated, but a full description with example of before/after implements.

Because if we just write a line, indeed the `@deprecated` line would be enough. :wink: 

Just see the Symfony upgrades guide for inspiration, it's quite well formatted.

File: UPGRADE-3.x.md [hunk 1]
```diff
@@ -9,3 +9,7 @@ UPGRADE FROM 3.0 to 3.1
 All files under the ``Tests`` directory are now correctly handled as internal test classes. 
 You can't extend them anymore, because they are only loaded when running internal tests. 
 More information can be found in the [composer docs](https://getcomposer.org/doc/04-schema.md#autoload-dev).
+
+### Deprecated
+
+`$container` property in `Security/SessionDownloadStrategy` is deprecated. Use `SessionInterface` `$session` instead.
```
Comment (Soullivaneuh) [inline UPGRADE-3.x.md]: Please add example of before/after implementation. Looks symfony upgrade guides for inspirations.

Comment (ahmetakbn) [inline UPGRADE-3.x.md]: is it ok like this @Soullivaneuh ?

Comment (Soullivaneuh) [inline UPGRADE-3.x.md]: Looks ok but I'm not an expert of media-bundle so I don't know if the example is enough.

cc @sonata-project/contributors for review.

Comment (OskarStark) [inline UPGRADE-3.x.md]: almost...

File: UPGRADE-3.x.md [hunk 2]
```diff
@@ -9,3 +9,19 @@ UPGRADE FROM 3.0 to 3.1
...
+
+Before:
+
+```php
+    $downloadStrategy = new SessionDownloadStrategy($translator, $container, $times);
```
Comment (OskarStark) [inline UPGRADE-3.x.md]: hmm, I am missing the typehint here, maybe its better so show the __construct() method with their typehints here to see the difference? @greg0ire

Comment (greg0ire) [inline UPGRADE-3.x.md]: I think people know what `$container` and `$session` probably refer to ;)

Comment (OskarStark) [inline UPGRADE-3.x.md]: ok, then lets merge as it is! 👍

**Comment by OskarStark** [comment]
Thank you @ahmetakbn 👍

**Pull request merged**